### DISCUSSION TITLE
It is bad practice to add headers to a response redirect?

### DISCUSSION THREAD

**Discussion by eduardfossas** [vercel/next.js#66077] [Help]
### Summary

Hello, I have seen that NextResponse.redirect does not allow to add headers. It is bad practice? and what will be the best solution to add the using a rewrite?

**Comment by icyJoseph** [comment]
You can make your own response with `new Response`, right?

Though, keep in mind that, if the browser follows the redirect Location header, it doesn't carry over other headers into the followed location request.
